### ISSUE TITLE
Cannot read the files written by the xlsx Haskell library.

### ISSUE THREAD

**Issue by stla** [hadley/readxl#225]
Hello, 

This post follows issue https://github.com/qrilka/xlsx/issues/74. The function `readxl::read_excel` can't read the XLSX files created by the Haskell library `xlsx`. 

Could I send you such files in order that you take a look ? Or what I could I do ? I'm ignorant in C, I don't know how to locate the problem.

**Comment by stla** [comment]
The error in R:
```
> readxl:::read_xlsx_(path, 0, TRUE, NULL, "")
Error in readxl:::read_xlsx_(path, 0, TRUE, NULL, "") : 
  Skipped over all data
```

**Comment by stla** [comment]
Stupid of me !! It works !! There are no column names and I set `colnames` to `TRUE` !
Sorry !

**Comment by stla** [comment]
However there's still a problem with another file. I'm busy now, I will reopen this issue later, perhaps.

**Comment by stla** [comment]
I've found the problem. I had a `styles.xml` like this:

`<cellXfs count="2"><xf fontId="0" fillId="0" borderId="0"/><xf ..............`

The first (or rather the 0-th) cellXf has no number format id. In this case the R session totally crashes when we run `readxl::read_excel` !

It works after setting a number format id: 

`<cellXfs count="2"><xf numFmtId="0" fontId="0" fillId="0" borderId="0"/><xf ..............`

However the file without numFmtId in the 0-th cellXf is valid for Excel.

I reopen the issue. I let you decide if this requires a fix. Of course I can set a numFmtId with Haskell.

**Comment by jennybc** [comment]
Can you provide a full example xlsx with the property described above?

**Comment by stla** [comment]
Yep, nice request. This will help to confirm this the one and only one cause. I'll do it later. Is there a nice way to share a file ? Otherwise there's dropbox anyway.

**Comment by stla** [comment]
No need to share a file. Try this:

```
#### set a working directory ####
wd <- "~/Rtests/Excel"
setwd(wd)

#### take a xlsx file ####
xlsx <- system.file("extdata/datasets.xlsx", package = "readxl")

#### unzip it ####
unzip(xlsx, exdir="./test")

#### remove the numFmtId in the O-th cellXfs
library(xml2)
xml_as_list <- as_list(read_xml("./test/xl/styles.xml"))
attr(xml_as_list$cellXfs$xf, "numFmtId") <- NULL
# note: you can also remove cellStyleXfs if you want
  # #xml_as_list$cellStyleXfs <- NULL
xml <- as_xml_document(list(styleSheet = xml_as_list))
write_xml(xml, "./test/xl/styles.xml")

#### zip ####
setwd("./test")
zip("../newxlsx.xlsx", list.files(recursive=TRUE, all.files=TRUE))
setwd("..")

#### check that newxlsx.xlsx can be open in Excel without issue ####

#### try to read it with readxl... BOOM!! ####
readxl::read_excel("newxlsx.xlsx")

```

**Comment by jennybc** [comment]
Fixed by 14d680d5e8. At least using an xlsx altered as described above. You might want to verify with an actual xlsx written directly by the Haskell library you're using.

**Comment by stla** [comment]
Thank you. 
There's no xlsx written "directly" by this Haskell library. The user has to define every cell, every number format, etc.
